examples/send_image.py

```diff
@@ -0,0 +1,177 @@
+        "info": {
+            "size": file_stat.st_size,
+            "mimetype": mime_type,
+            "thumbnail_info": None,  # TODO
+            "w": width,  # width in pixel
+            "h": height,  # height in pixel
+            "thumbnail_url": None,  # TODO
+        },
+        "msgtype": "m.image",
+        "url": resp.content_uri,
+    }
+
+    try:
+        await client.room_send(
+            room_id,
+            message_type="m.room.message",
+            content=content
+        )
+        print("Image was sent successfully")
+    except Exception:
+        print(f"Image send of file {image} failed.")
+
+
+async def main() -> None:
+    # If there are no previously-saved credentials, we'll use the password
+    if not os.path.exists(CONFIG_FILE):
+        print("First time use. Did not find credential file. Asking for "
+              "homeserver, user, and password to create credential file.")
+        homeserver = "https://matrix.example.org"
+        homeserver = input(f"Enter your homeserver URL: [{homeserver}] ")
+
+        if not (homeserver.startswith("https://")
+                or homeserver.startswith("http://")):
+            homeserver = "https://" + homeserver
+
+        user_id = "@user:example.org"
+        user_id = input(f"Enter your full user ID: [{user_id}] ")
+
+        device_name = "matrix-nio"
+        device_name = input(f"Choose a name for this device: [{device_name}] ")
+
+        client = AsyncClient(homeserver, user_id)
+        pw = getpass.getpass()
+
+        resp = await client.login(pw, device_name=device_name)
+
+        # check that we logged in succesfully
+        if (isinstance(resp, LoginResponse)):
+            write_details_to_disk(resp, homeserver)
+        else:
+            print(f"homeserver = \"{homeserver}\"; user = \"{user_id}\"")
+            print(f"Failed to log in: {resp}")
+            sys.exit(1)
+
+        print(
+            "Logged in using a password. Credentials were stored.",
+            "Try running the script again to login with credentials."
+        )
+
+    # Otherwise the config file exists, so we'll use the stored credentials
+    else:
+        # open the file in read-only mode
+        with open(CONFIG_FILE, "r") as f:
+            config = json.load(f)
+            client = AsyncClient(config['homeserver'])
+
+            client.access_token = config['access_token']
+            client.user_id = config['user_id']
+            client.device_id = config['device_id']
+
+        # Now we can send messages as the user
+        room_id = "!myfavouriteroomid:example.org"
+        room_id = input(f"Enter room id for image message: [{room_id}] ")
+
+        image = "exampledir/samplephoto.jpg"
+        image = input(f"Enter file name of image to send: [{image}] ")
+
+        await send_image(client, room_id, image)
+        print("Logged in using stored credentials. Sent a test message.")
+
+    # Either way we're logged in here, too
```

Comment:
```suggestion
    # Close the client connection after we are done with it.
```

Reply:
thanks for the feedback, your suggestion is implemented in commit # 2 in c8d4ad4403afb97e87c9be4a2af633007d8df360